ocs_ci/ocs/resources/cloud_manager.py

```diff
@@ -0,0 +1,305 @@
+            self.access_key = credentials.access_key
+            self.secret_key = credentials.secret_key
+        bs_secret_data = templating.load_yaml(constants.MCG_BACKINGSTORE_SECRET_YAML)
+        bs_secret_data['metadata']['name'] += f'-client-secret'
+        bs_secret_data['metadata']['namespace'] = config.ENV_DATA['cluster_namespace']
+        bs_secret_data['data']['AWS_ACCESS_KEY_ID'] = base64.urlsafe_b64encode(
+            self.access_key.encode('UTF-8')
+        ).decode('ascii')
+        bs_secret_data['data']['AWS_SECRET_ACCESS_KEY'] = base64.urlsafe_b64encode(
+            self.secret_key.encode('UTF-8')
+        ).decode('ascii')
+        self.secret = create_resource(**bs_secret_data)
+
+    def internal_create_uls(self, name, region=None):
+        """
+        Creates the Underlying Storage using the S3 API
+
+        Args:
+           name (str): The Underlying Storage name to be created
+           region (str): The region to create the Underlying Storage, if none will create at
+           `us-east-1` IMPORTANT!!! note that `us-east-1` will cause an error if used since it is
+           the default region for aws
+
+        """
+        if region is None:
+            self.client.create_bucket(Bucket=name)
+        else:
+            self.client.create_bucket(
+                Bucket=name,
+                CreateBucketConfiguration={
+                    'LocationConstraint': region
+                }
+            )
+
+    def internal_delete_uls(self, name):
+        """
+        Deletes the Underlying Storage using the S3 API
+
+        Args:
+           name (str): The Underlying Storage name to be deleted
+
+        """
+        # Todo: rename client to resource (or find an alternative)
+        self.client.meta.client.delete_bucket_policy(
+            Bucket=name
+        )
+        sample = TimeoutSampler(
+            timeout=30, sleep=3, func=self.check_uls_deletion,
+            name=name
+        )
+        if not sample.wait_for_func_status(result=True):
+            logger.error(
+                f'Deletion of Underlying Storage {name} timed out. Unable to delete {name}'
+            )
+            raise TimeoutExpiredError
+        else:
+            logger.info(f'Underlying Storage {name} deleted successfully')
+
+    def check_uls_deletion(self, name):
+        """
+        Try to delete Underlying Storage by name
+
+        Args:
+            name (str): the Underlying Storage name
+
+        Returns:
+            bool: True if deleted successfully
+
+        """
+        if self.verify_uls_exists(name):
+            try:
+                self.client.Bucket(name).objects.all().delete()
+                self.client.Bucket(name).delete()
+                return True
+            except ClientError:
+                logger.info(f'Deletion of Underlying Storage {name} failed. Retrying...')
+                return False
+        else:
+            return True
```

Comment:
Please log the incident (`ULS does not exist, and cannot be deleted` and raise an exception)